From 744f26db1af4d3deb652f4ab59971556942a4f5e Mon Sep 17 00:00:00 2001
From: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
Date: Thu, 16 Mar 2023 17:05:10 +0530
Subject: [PATCH] Fix Build error for health HAL

FAILED: out/soong/build.ninja
cd "$(dirname "out/host/linux-x86/bin/soong_build")" && BUILDER="$PWD/$(basename "out/host/linux-x86/bin/soong_build")" && cd / && env -i  "$BUILDER"     --top "$TOP"     --soong_out "out/soong"     --out "out"     -o out/soong/build.ninja --bazel-mode --globListDir build --globFile out/soong/globs-build.ninja -t -l out/.module_paths/Android.bp.list --available_env out/soong/soong.environment.available --used_env out/soong/soong.environment.used.build Android.bp
error: vendor/intel/hardware/interfaces/health/aidl/Android.bp:85:1: module "libhealth_aidl_impl.intel" variant "android_recovery_x86_64_silvermont_static": depends on multiple versions of the same aidl_interface: android.hardware.health-V1-ndk-source, android.hardware.health-V2-ndk-source

Tracked-On: NA
Signed-off-by: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
---
 health/aidl/Android.bp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/health/aidl/Android.bp b/health/aidl/Android.bp
index 01456c6..7c38c14 100644
--- a/health/aidl/Android.bp
+++ b/health/aidl/Android.bp
@@ -20,7 +20,7 @@ cc_defaults {
         "libcutils",
         "liblog",
         "libutils",
-        "android.hardware.health-V1-ndk",
+        "android.hardware.health-V2-ndk",
 
         // TODO(b/177269435): remove when BatteryMonitor works with AIDL HealthInfo.
         "libhidlbase",
@@ -39,7 +39,7 @@ cc_defaults {
     name: "libhealth_aidl_charger_defaults.intel",
     shared_libs: [
         // common
-        "android.hardware.health-V1-ndk",
+        "android.hardware.health-V2-ndk",
         "libbase",
         "libcutils",
         "liblog",
-- 
2.17.1

